flink-table/flink-sql-parser-hive/pom.xml

```diff
@@ -56,6 +56,12 @@ under the License.
 			<artifactId>guava</artifactId>
 			<version>${guava.version}</version>
 			<scope>provided</scope>
+			<exclusions>
```

Comment:
Just to double check: Do we really need an exclusion if the dep is already set to provided?

Reply:
🙈, you're right, thanks
removed it from here and flink-sql-parser

Reply:
yes that would be a reason to add the excelusion, better to triple check what maven contains in the `tree` and what is contained in the final jar files in the end.

Reply:
for `flink-table-planner` `mvn dependency:tree` gives
```
...
[INFO] +- org.apiguardian:apiguardian-api:jar:1.1.2:compile
...
```
compile is expected because of `org.apache.calcite.sql.validate.SqlValidatorImpl` where there are some apiguardian's annotations

for `flink-sql-parser` `mvn dependenc:tree` gives
```
...
[INFO] +- org.junit.vintage:junit-vintage-engine:jar:5.8.1:test
[INFO] |  +- org.junit.platform:junit-platform-engine:jar:1.8.1:test
[INFO] |  +- junit:junit:jar:4.13.2:test
[INFO] |  \- org.apiguardian:apiguardian-api:jar:1.1.2:test
...
```

fpr `flink-sql-parser-hive` `mvn dependency tree` gives
```
...
[INFO] +- org.junit.vintage:junit-vintage-engine:jar:5.8.1:test
[INFO] |  +- org.junit.platform:junit-platform-engine:jar:1.8.1:test
[INFO] |  +- junit:junit:jar:4.13.2:test
[INFO] |  \- org.apiguardian:apiguardian-api:jar:1.1.2:test

...
```
And neither jars from these 3 contains apiguardian

Reply:
just noted here about future rebase
Also after rebase to master some exclusions should be applied for flink-sql-client after this change https://issues.apache.org/jira/browse/FLINK-29486

Reply:
applied this change since rebased